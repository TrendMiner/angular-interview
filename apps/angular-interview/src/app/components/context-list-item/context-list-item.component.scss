.item {
  &__name {
    font-weight: bold;
  }

  &__type {
    text-transform: uppercase;
  }
}
