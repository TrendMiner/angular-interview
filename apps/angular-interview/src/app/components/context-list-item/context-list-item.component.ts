import { Component, Input } from '@angular/core';
import { ContextItem } from '../../domain/models/context-item.model';

//TODO: #1 Make sure a context item is rendered correctly

@Component({
  selector: 'tm-context-list-item',
  template: `
    <div class='item'>
      -
    </div>`,
  styleUrls: ['./context-list-item.component.scss']
})
export class ContextListItemComponent {
  @Input() contextItem: ContextItem;
}
