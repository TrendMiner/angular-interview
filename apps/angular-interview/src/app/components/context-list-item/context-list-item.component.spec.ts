import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
import {By} from '@angular/platform-browser';
import {ContextListItemComponent} from './context-list-item.component';
import { ContextItem } from '../../domain/models/context-item.model';

describe('ContextListItemComponent', () => {
  let fixture: ComponentFixture<ContextListItemComponent>;
  let component: ContextListItemComponent;

  const ITEM_SELECTOR = By.css('.item');

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [],
      declarations: [
        ContextListItemComponent
      ],
      providers: [],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ContextListItemComponent);
    component = fixture.debugElement.componentInstance;
  });

  it('should display the name and type', () => {
    component.contextItem =
      {
        name: 'Gold',
        type: 'Batch'
      } as ContextItem;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
