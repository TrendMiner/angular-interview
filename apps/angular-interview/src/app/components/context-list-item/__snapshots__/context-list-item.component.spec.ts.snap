// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextListItemComponent should display the name and type 1`] = `
<tm-context-list-item
  contextItem={[Function Object]}
>
  <div
    class="item"
  >
    <span
      class="item__name"
    >
      Gold
    </span>
     - 
    <span
      class="item__type"
    >
      Batch
    </span>
  </div>
</tm-context-list-item>
`;
